python/lsst/daf/butler/datastores/fileDatastore.py

```diff
@@ -104,7 +90,7 @@ class _IngestPrepData(Datastore.IngestPrepData):
         Files to be ingested by this datastore.
     """
 
-    def __init__(self, datasets: List[FileDataset]):
+    def __init__(self, datasets: list[FileDataset]):
```

Comment:
```suggestion
    def __init__(self, datasets: Iterable[FileDataset]):
```
(+ associated doc change)